packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -15,9 +17,10 @@ import '../runner/flutter_command.dart';
 class UpdatePackagesCommand extends FlutterCommand {
   UpdatePackagesCommand({ this.hidden: false }) {
     argParser.addFlag(
-      'upgrade',
-      help: 'Ignores pubspec.lock and retrieves newer versions of packages.',
-      defaultsTo: false
+      'force-upgrade',
+      help: 'Attempt to update all the dependencies to their latest versions.\n'
+            'This will actually modify the pubspec.yaml files in your checkout.',
+      defaultsTo: false,
```

Comment:
This command could also have a dry run mode, which would allow you to write a test that does not modify the pubspecs.

Reply:
Since the pubspecs are under git's control, every run is a "dry run" really.